src/jupyterlabcontroller/models/v1/domain/tag.py

```diff
@@ -0,0 +1,534 @@
+"""Classes to hold all the semantic data and metadata we can extract from a
```

Comment:
Yeah, all of this is kind of in transition because I've mostly airlifted it over from cachemachine and wanted to reinvent as few wheels as possible.  It seemed to make sense there to make the tags mostly dataclasses.  I *think* where I want to go with this is the output of consolidate_tags: when it's done, it has one preferred tag for a given digest, which maps to a display name.  That image appears once in the all-tags image, under the "best" tag (which is basically just a sort function on tag type: Release is better than Weekly is better than Daily, etc.).

Reply:
I've kept the Tag as a model, but it's not a domain model.  RSPStandaloneTag is everything that can be determined from the tag text; RSPTag is that augmented with stuff we glue in from elsewhere, like the digest, the Docker repo path, and what nodes it's on.